src/components/profile/Complete/index.js

```diff
@@ -0,0 +1,272 @@
+
+  useEffect(() => {
+    setBackMsg(msg);
+  }, [msg]);
+
+  setTimeout(() => {
+    setBackMsg('');
+  }, 15000);
+
+  useEffect(() => {
+    setBackErrors(backErrors);
+  }, [backErrors]);
+
+  setTimeout(() => {
+    setBackErrors('');
+  }, 15000);
+
+  const validate = (state) => {
+    let errors = {};
+    if (!state.gender) {
+      errors.gender = 'Gender is required';
+    }
+    if (!state.birthdate) {
+      errors.birthdate = 'Date of birth is required';
+    }
+    if (!state.language) {
+      errors.language = 'Language is required';
+    }
+    if (!state.currency) {
+      errors.currency = 'Currency is required';
+    }
+    if (!state.country) {
+      errors.country = 'Country is required';
+    }
+    if (!state.department) {
+      errors.department = 'Department is required';
+    }
+    if (!state.linemanager) {
+      errors.linemanager = 'Line manager is required';
+    }
+    return errors;
+  };
+  const onChange = (e) => {
+    const { name, value } = e.target;
+    state[name] = value;
+    newState(state);
+  };
+  const onSubmit = (e) => {
+    e.preventDefault();
+    setErrors(validate(state));
+    setIsSubmitting(true);
+  };
+  useEffect(() => {
+    if (Object.keys(errors).length === 0 && isSubmitting) {
+      dispatch(completeProfile(state));
+    }
+  }, [errors]);
+
+  const profile = (
+    <Row>
+      <Col md='6' className='p-3 background-form'>
+        <h3 className='text-center text-color'>Complete Your Profile</h3>
+        <Form onSubmit={onSubmit}>
+          {backMsg ? (
+            <Alert color='success' className='text-center'>
+              {backMsg}
+            </Alert>
+          ) : (
+            ''
+          )}
+          {getBackErrors ? (
+            <Alert color='danger' className='text-center'>
+              {getBackErrors}
+            </Alert>
+          ) : (
+            ''
+          )}
+          <Row className='mb-3'>
+            <Col md='6'>
+              <FormGroup>
+                <Label className='text-color'>GENDER</Label>
+                <Input
+                  type='text'
+                  name='gender'
+                  id='gender'
+                  onChange={onChange}
+                  className={errors.gender ? `border-error` : ''}
+                  placeholder='eg: Male'
+                />
+
+                {errors.gender ? (
+                  <Alert color='danger' className='back-error'>
+                    {errors.gender}
+                  </Alert>
+                ) : (
+                  ''
+                )}
+              </FormGroup>
+              <FormGroup>
+                <Label className='text-color'>LANGUAGE</Label>
+                <Input
+                  type='text'
+                  name='language'
+                  id='language'
+                  onChange={onChange}
+                  className={errors.gender ? `border-error` : ''}
+                  placeholder='eg: English'
+                />
```

Comment:
Kindly, add a drop-down which contains the languages we support

Reply:
solved